src/wallet/wallet.ts

```diff
@@ -122,18 +136,130 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
+   * @inner
+   */
+  static getAuthXPubKeyFromSeed(seed: string, options: { passphrase?: string, networkName?: string } = {}): string {
+    const methodOptions = Object.assign({
+      passphrase: '',
+      networkName: 'mainnet',
+    }, options);
+
+    const xpriv = walletUtils.getXPrivKeyFromSeed(seed, methodOptions);
+    const privkey = HathorWalletServiceWallet.deriveAuthPrivateKey(xpriv);
+
+    return privkey.xpubkey;
+  }
+
+  /**
+   * Derive private key from root to the auth specific purpose derivation path
+   *
+   * @param {HDPrivateKey} xpriv The wallet's root xpriv
+   *
+   * @return {HDPrivateKey} Derived private key at the auth derivation path
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  static deriveAuthPrivateKey(xpriv: bitcore.HDPrivateKey): bitcore.HDPrivateKey {
+    return xpriv.deriveNonCompliantChild(WALLET_SERVICE_AUTH_DERIVATION_PATH);
+  }
+
+  /**
+   * Encrypt and store wallet data on storage using the wallet's PIN code
+   *
+   * @param {string} pin Pin code to use as password to encrypt the auth key
+   * @param {string} mainKey xpriv of the change level derivation to sign inputs
+   * @param {string} authKey xpriv of the auth specific purpose derivation path
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  static initializeAccessData(pin: string, mainKey: string, authKey: string) {
+    const initialAccessData = wallet.getWalletAccessData() || {};
+
+    const encryptedAuthKey = wallet.encryptData(authKey, pin);
+    const encryptedMainKey = wallet.encryptData(mainKey, pin);
+
+    initialAccessData['authKey'] = encryptedAuthKey.encrypted.toString();
+    initialAccessData['mainKey'] = encryptedMainKey.encrypted.toString();
+
+    wallet.setWalletAccessData(initialAccessData);
+  }
+
   /**
    * Start wallet: load the wallet data, update state and start polling wallet status until it's ready
    *
+   * @param {Object} optionsParams Options parameters
+   *  {
+   *   'pinCode': PIN to encrypt the auth xpriv on storage
+   *  }
+   *
    * @memberof HathorWalletServiceWallet
    * @inner
    */
-  async start() {
+  async start({ pinCode }) {
+    if (!this.seed) {
+      throw new Error('Seed should be in memory when starting the wallet.');
+    }
+
+    if (!pinCode) {
+      throw new Error('Pin code is required when starting the wallet.');
+    }
+
     this.setState(walletState.LOADING);
+
+    const xpriv = walletUtils.getXPrivKeyFromSeed(this.seed, {passphrase: this.passphrase, networkName: this.network.name});
     const xpub = walletUtils.getXPubKeyFromSeed(this.seed, {passphrase: this.passphrase, networkName: this.network.name});
+    const authXpub = HathorWalletServiceWallet.getAuthXPubKeyFromSeed(this.seed, {
+      passphrase: this.passphrase,
+      networkName: this.network.name,
+    });
+
+    const now = new Date();
+    const timestampNow = Math.floor(now.getTime() / 1000);
+    const walletId = crypto.Hash.sha256sha256(Buffer.from(xpub)).toString('hex');
```

Comment:
Refactor to a separate method called `getWalletIdFromXPub(xpub)`?

Reply:
Sounds good!

Refactored in 695f96c333624d48f82d0e1d975bd2515e001378

Reply:
Great! Thanks!